src/base/LruMap.h

```diff
@@ -148,39 +169,47 @@ LruMap<EntryValue, EntryCost>::add(const char *key, EntryValue *t)
     if (memLimit() == 0)
         return false;
 
-    index.push_front(new Entry(key, t));
+    auto e = new Entry(key, t, ttl);
+    memUsed_ += e->memCost() + e->key.length() + sizeof(Entry) + sizeof(MapPair);
+    index.push_front(e);
     storage.insert(MapPair(key, index.begin()));
 
     ++entries_;
+
+    // if over memory limit, start purging
+    while (memUsed_ > memLimit_)
```

Comment:
I believe your recent changes follow the suggestions I outlined in this stale comment so it can probably be considered addressed.